contracts/Recruitment.sol

```diff
@@ -72,7 +84,143 @@ contract Recruitment {
+    refereeList[msg.sender] = referee;
+  }
+
+  /**
+    * @notice Getter function of Referee.
+    * @param wallet The wallet address of a referee to be fetched from the mapping
+    * @return Referee The type of FrontDoorStruct.Referee 
+  */
+  function getReferee(address wallet) external view returns(FrontDoorStructs.Referee memory) {
+    return refereeList[wallet];
+  }
+
+  /**
+    * @notice Registers a job
+    * @param bounty The amount of bounty set by the job poster.
+    * @param maxSalary The amount of maximum salary.
+    * @param minSalary The amount of minimum salary.
+    * @param companyName The name of company.
+    * @param location The location of the company.
+    * @param role The role for the job.
+    * @param jobURL The web URL of the job description.
+  */
+  function registerJob(uint256 bounty, uint256 maxSalary, uint256 minSalary, string memory companyName, string memory location, string memory role, string memory jobURL) external {
+    uint256 jobId = jobIdCounter.current();
+    FrontDoorStructs.Job memory job = FrontDoorStructs.Job(jobId, bounty, maxSalary, minSalary, companyName, location, role, jobURL);
+    jobList[jobId] = job;
+    jobIdCounter.increment();
+  }
+
+  /**
+    * @notice Getter function of Job.
+    * @param jobId The ID of a job to fetch from the mapping
+    * @return Job The type of FrontDoorStruct.Job 
+  */
+  function getJob(uint256 jobId) external view returns(FrontDoorStructs.Job memory){
+    return jobList[jobId];
+  }
+
+  /**
+    * @notice Registers a referral
+    * @param jobId The job ID already registered with the contract.
+    * @param refereeWallet The address of referee.
+    * @param refereeMail The email of the referee.
+  */
+  function registerReferral(uint256 jobId, address refereeWallet, string memory refereeMail) external {
+    FrontDoorStructs.Referee memory referee = refereeList[refereeWallet];
+    FrontDoorStructs.Referrer memory referrer = referrerList[msg.sender];
+    FrontDoorStructs.Job memory job = jobList[jobId];
+
+    if (referee.wallet == address(0)) {
+      referee = FrontDoorStructs.Referee(refereeWallet, refereeMail, 0);
+      refereeList[refereeWallet] = referee;
+    }
+
+    FrontDoorStructs.Referral memory referral = FrontDoorStructs.Referral(referralCounter.current(), false, referrer, referee, job);
+
+    referralIndex[msg.sender].push(referralCounter.current());
+    referralList[referralCounter.current()] = referral;
+    referralCounter.increment();
+
+  }
+
+  /**
+    * @notice Confirmation of a referral by the msg.sender (if the msg.sender is the referee)
+    * @param refId The referral ID already recorded in the smart contract.
+  */
+  function confirmReferral(uint256 refId) external {
+    
+    FrontDoorStructs.Referral memory referral = referralList[refId];
+
+    if (referral.referee.wallet != msg.sender) revert Errors.SenderIsNotReferee(); // The msg.sender should be the referee.
```

Comment:
Here and continuing in the light of the new confirmation updated flow we cannot hav e this revert.

Reply:
okay, then, since it is an external/public function, anyone can amend the data (i.e., confirm a referral, even if that referral does not belong to them). This doesn't make sense to me. How are we going to ensure that a referee can only confirm their own referrals?

Reply:
So, this check remains there, as it checks the `emailHash` now, instead wallet.